src/actions/init/init.actions.ts

```diff
@@ -0,0 +1,212 @@
+import { checkNodeVersion } from '../../utils/utilities.helper';
+import { Action } from '../abstract.action';
+import { InitActionOptions } from './init.types';
+import { MongoClient } from 'mongodb';
+import { PackageManager, PackageManagerName, installDependencies } from 'nypm';
+
+import fsExtra from 'fs-extra';
+import { $, execaCommand } from 'execa';
+import fs from 'fs';
+import inquirer from 'inquirer';
+import ora from 'ora';
+import path from 'path';
+import { changeDirectory } from '../../utils/files.helper';
+import { printWithMantisGradient } from '../../utils/prettyPrint.helper';
+
+type PMTypePromptResult = Pick<PackageManager, 'name'>;
+
+export default class InitAction extends Action {
+  private options: InitActionOptions;
+
+  constructor(options: InitActionOptions) {
+    super('[INIT-ACTION]', { checkMantisProject: false });
+    this.options = options;
+  }
+
+  promptForPackageManagerSelect = async (): Promise<PMTypePromptResult> => {
+    const temp = await inquirer.prompt<PMTypePromptResult>({
+      type: 'list',
+      name: 'name' satisfies keyof PMTypePromptResult,
+      message: "Select the package manager you'd like to use:",
+      choices: ['npm', 'bun', 'pnpm', 'yarn'] satisfies PackageManagerName[],
+    });
+    return temp;
+  };
+
+  installDependenciesWithMessage = async (workspacePath: string) => {
+    const pm = await this.promptForPackageManagerSelect();
+    const spinner = ora('Installing dependencies').start();
+    await installDependencies({
+      packageManager: pm.name,
+      cwd: workspacePath,
+      silent: true,
+    });
+    spinner.succeed('Installed dependencies');
+    return pm;
+  };
+
+  async promptForMissingValues() {
+    type PromptResult = {
+      workspaceName: string;
+      mongoUrl: string;
+      createMobile: boolean;
+    };
+    const answers = await inquirer.prompt<PromptResult>([
+      {
+        type: 'input',
+        name: 'workspaceName',
+        message: 'Enter the name of the workspace to create:',
+        default: 'mantis',
+        when: () => !this.options.workspaceName,
+        validate: async (value) => {
+          if (value.trim().length <= 0)
+            return 'Workspace name must not be empty';
+          if (await fsExtra.pathExists(value)) {
+            return `Workspace already exists: ${value}`;
+          }
+
+          return true;
+        },
+      },
+      {
+        type: 'input',
+        name: 'mongoUrl',
+        message:
+          "Enter the MongoDB URI (default is 'mongodb://localhost:27017/mantis'):",
+        default: 'mongodb://localhost:27017/mantis',
```

Comment:
A few things to note.

Could we update the prompt to be clear that we will set up the db for the user? The original prompt used the following options:

```
      {
        name: 'Create a local development mongo db for me (default)',
        value: 'local',
      },
      {
        name: 'Provide a mongo connection string',
        value: 'dbUrl',
      },
```

Secondly, can we make the db unique to the project? We wouldn't want every mantis project the user creates to connect to the same db.